src/v/model/fundamental.h

```diff
@@ -229,7 +229,7 @@ struct topic_partition {
     model::partition_id partition;
 
     bool operator==(const topic_partition& other) const {
-        return topic == other.topic && partition == other.partition;
+        return partition == other.partition && topic == other.topic;
```

Comment:
Comparing the partition first helps a lot when `==` returns false, since large numbers of partitions and modest number of topics are common (especially in the context of a single fetch request, which is often to N partitions from 1 topic), so it is common that the topic matches but the partition doesn't, so try compare the partition first. Furthermore, this comparison is relatively cheap (just a single 4-byte compare) vs topic which is a string compare, so even at equal probabilities of "not match" this would be a win.